src/nbl/video/COpenGLCommandPool.h

```diff
@@ -47,6 +64,248 @@ class COpenGLCommandPool final : public IGPUCommandPool
         core::CMemoryPool<core::GeneralpurposeAddressAllocator<uint32_t>,core::default_aligned_allocator,false,uint32_t> mempool;
 };
 
+class NBL_FORCE_EBO COpenGLCommandPool::ICommand
+{
+public:
+    virtual void operator() (IOpenGL_FunctionTable* gl, SOpenGLContextLocalCache* ctxLocal, uint32_t ctxid, const system::logger_opt_ptr logger, const COpenGLCommandBuffer* executingCmdbuf) = 0;
+
+    using base_cmd_t = void;
+
+protected:
+    template <typename CRTP, typename... Args>
+    static uint32_t end_offset(const Args&... args)
+    {
+        using other_base_t = typename CRTP::base_cmd_t;
+        if constexpr (std::is_void_v<other_base_t>)
+            return sizeof(CRTP);
+        else
+            return other_base_t::calc_size(args...) - sizeof(other_base_t) + sizeof(CRTP);
+    }
```

Comment:
if you accept my QoL idea, all this complex stuff won't be needed.

Reply:
Removed. Resolved.